### ISSUE TITLE
Germline SV calls from matched tumour and normal SV calling

### ISSUE THREAD

**Issue by tgong1** [PapenfussLab/gridss#374]
Hi Daniel,

I did a matched tumour and normal SV calling and used gridss_somatic_filter.R to obtain the somatic SV calls. I'm wondering is it possible to have the germline SV calls using the raw vcf file output?

Thank you,
Tingting

**Comment by d-cameron** [comment]
Yes. The raw VCF file contains all call from variants found in the normal and/or the tumour. Calls found in the normal will either be germline, or noise. Unfortunately, at this stage I don't have a nice clean script to separate out the likely TP germline calls from the noise. Performance on germline SVs under 100kbp is pretty good (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6642177/) but the larger SVs are almost all going to be noise (e.g. a FP interchromsomal SV between NOTCH2 and a NOTCH2 pseudogene in pretty much every sample due to hg19 having low penetrance SNVs in the region of homology).

We have a PhD student working on a ML approach to filter with some nice results but that work is still in development so we don't have anything release quality yet.

TLDR: current approach is to filter to high confidence ,<100kbp, and above QUAL threshold in the normal.

**Comment by DarioS** [comment]
Would it be better to refer to them as _non-reference_ variants instead? The majority of the reference genome is generated from one person's DNA, code-named RPC-11, and is unrelated to the person being sequenced. It is different to having the parents also sequenced - a biology textbook definition of _germline_.

**Comment by tgong1** [comment]
Hi Daniel,

I was obtaining germline SVs by extracting high confidence SVs in the normal, using the following commands in R. Please let me know if this is not a proper way or there is better way for it.
vcf <- vcf[vcf@fixed$QUAL >= 500,]
vcf <- vcf[VariantAnnotation::geno(vcf)$QUAL[,2] < 0.05 * vcf@fixed$QUAL,]###normal is the first INPUT, tumor is the second INPUT

By looking at the resulted VCF, I found some results which not sure if that is supposed to see. The following example SV has QUAL score of 2537.34, however, both QUAL for normal and tumour are 0. Is this supposed to see? What does QUAL of 0 mean?

chr1	99117	gridss0f_314b	G	GACAGTGTAAAAAACTTTCACCTGGGGTGGAGTGCATTGTGGGAAACACGACTAACTGCAAATTATGCCACCCAGGTTAAAGTGATTAGACAGCCAAAGCCCCCCAAGGAGGGGGG.	2537.34	PASS	AS=0;ASC=1X;ASQ=0.00;ASRP=0;ASSR=0;BA=4;BANRP=0;BANRPQ=0.00;BANSR=0;BANSRQ=0.00;BAQ=578.15;BASRP=3;BASSR=21;BEID=asm0-604,asm0-613,asm0-620,asm0-636;BEIDH=-1,-1,-1,-1;BEIDL=0,0,0,0;BQ=2537.34;BSC=80;BSCQ=1833.86;BUM=5;BUMQ=125.32;BVF=82;CAS=0;CASQ=0.00;CQ=1111.17;EVENT=gridss0f_314;IC=0;IQ=0.00;RAS=0;RASQ=0.00;REF=255;REFPAIR=255;RP=0;RPQ=0.00;SB=1.0;SC=1X;SR=0;SRQ=0.00;SVTYPE=BND;VF=0	GT:ASQ:ASRP:ASSR:BANRP:BANRPQ:BANSR:BANSRQ:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF	.:0.00:0:0:0:0.00:0:0.00:184.12:0:8:753.87:24:541.43:1:28.32:24:0.00:0:0.00:0.00:0.00:67:76:0:0.00:0:0.00:0	.:0.00:0:0:0:0.00:0:0.00:394.03:3:13:1783.46:56:1292.43:4:97.00:58:0.00:0:0.00:0.00:0.00:188:179:0:0.00:0:0.00:0

Thank you,
Tingting

**Comment by d-cameron** [comment]
> What does QUAL of 0 mean?

The QUAL field is the overall variant score. For breakpoints this matches the `FORMAT` `QUAL` fields but for single breakend, it's the `FORMAT` `BQ` field. That is:

`FORMAT` `QUAL`: per sample break*point* support (split reads, discordant read pairs, breakpoint assemblies)
`FORMAT` `BQ`: per sample break*end* support (soft clipped reads, reads with unmapped mates, breakend assemblies)

Breakpoint will typically have non-zero `FORMAT` `QUAL` and `BQ` (since reads spanning the breakpoint by less then ~25bp will be aligned as soft-clipped reads, not split reads) but the overall `QUAL` score for breakpoints is determined only from the breakpoint support (not the breakend `BQ` support).

Breakends will always have zero `FORMAT` `QUAL`.

**Comment by d-cameron** [comment]
> Would it be better to refer to them as non-reference variants instead?

It would. GRIDSS variant calling is label agnostic and does general purpose multi-sample calling with no special treatment of any particular sample.  The somatic filtering script uses "normal" and "tumour" terminology as, as you have pointed out, it's the more techncially correct terminology.
